build/AzurePipelinesTemplates/WindowsAppSDK-BuildVSIX-Steps.yml

```diff
@@ -0,0 +1,236 @@
+- name: IsOneBranch
+  type: boolean
+  default: False
+- name: runStaticAnalysis
+  type: boolean
+  default: True
+- name: UseCurrentBuild
+  type: boolean
+  default: False 
+- name: OptionalVSIXVersion
+  # if blank, the project template will select the version matching the Windows App SDK
+  # nuget. If provided, it should be a 4-part dotted version ('1.2.3.4')
+  type: string
+  default: 'default'
+- name: EnableExperimentalVSIXFeatures
+  type: boolean
+  default: False
+
+steps:
+- ${{ if eq(parameters.UseCurrentBuild, 'True') }}:
+  - task: DownloadPipelineArtifact@2
+    displayName: 'Download WindowsAppSDK From Current Build'
+    inputs:
+      artifactName: 'WindowsAppSDK_Nuget_And_MSIX'
+      targetPath: '$(System.ArtifactsDirectory)'
+- ${{ else }}:
+  ###
+  # This step downloads the WindowsAppSDK NuGet package from a pipeline
+  # variables: OfficialPipelineID LatestOfficialBuildID
+  # for use in building the VSIX
+  - task: DownloadPipelineArtifact@2
+    displayName: 'Download WindowsAppSDK From Latest Nightly'
+    inputs:
+      artifactName: 'WindowsAppSDK_Nuget_And_MSIX'
+      targetPath: '$(System.ArtifactsDirectory)'
+      source: 'specific'
+      project: $(System.TeamProjectId)
+      pipeline: $(OfficialPipelineID) 
+      pipelineId: $(LatestOfficialBuildID) 
+    
+- task: PowerShell@2
+  name: ExtractWindowsAppSDKVersion
+  displayName: Extract WindowsAppSDKVersion
+  inputs:
+    targetType: 'inline'
+    script: |
+      $files = Get-ChildItem $(System.ArtifactsDirectory)
+      foreach ($file in $files) # Iterate through each package we restored in the directory
+      {
+        Write-Host "file:" $file.FullName
+        $nupkgPaths = Get-ChildItem $file.FullName -Filter "*.nupkg"
+
+        # Extract nupkg to access the nuspec
+        # The files in this directory does not contain the nuspec by default
+        foreach ($nupkgPath in $nupkgPaths)
+        {
+          Write-Host "nupkgPath:" $nupkgPath.FullName
+          $rename = $nupkgPath.Name + ".zip"
+          Rename-Item $nupkgPath.FullName $rename
+          $renamedFilePath = $nupkgPath.FullName + ".zip"
+          $dest = $file.FullName + "/contents"
+          Expand-Archive $renamedFilePath -Destination $dest
+        }
+
+        $nuspecPaths = Get-ChildItem $file.FullName -Recurse -Filter "*.nuspec"
+        foreach ($nuspecPath in $nuspecPaths)
+        {
+          Write-Host "Found Nuspecs"
+          [xml]$nuspec = Get-Content -Path $nuspecPath.FullName
+          if ($nuspec.package.metadata.id -eq 'Microsoft.WindowsAppSDK')
+          {
+            $version = $nuspec.package.metadata.version
+            Write-Host "Found " $version
+            Write-Host "##vso[task.setvariable variable=WindowsAppSDKPackageVersion]$version"
+            Write-Host "##vso[task.setvariable variable=WindowsAppSDKPackageVersion;isOutput=true]$version"
+            Exit 0
+          }
+        }
+      }
+      Exit 1
+
+- task: NuGetToolInstaller@1
```

Comment:
Do I understand it right that, from here on, the code is similar to of that in WindowsAppSDK-CreateVSIX-Steps.yml on the agg repo? If so, would it be feasible to refactor this file to only keep one copy of WindowsAppSDK-CreateVSIX-Steps.yml, say, on the config repo, to reduce the recurring burden of keeping 2 copies in sync?

Reply:
Oh yea I should have made this clear in the description. We are removing it from the Aggregator repo.